lib/metrics/StatsModel.js

```diff
@@ -88,10 +93,19 @@ class StatsModel extends StatsClient {
                 return cb(null, statsRes);
             }
 
-            statsRes.requests = this._zip(requests).map(arr =>
+            requests = this._zip(requests).map(arr =>
                 arr.reduce((acc, i) => acc + i));
-            statsRes['500s'] = this._zip(errors).map(arr =>
+            errors = this._zip(errors).map(arr =>
                 arr.reduce((acc, i) => acc + i));
+
+            // replace any existing values in arrays "requests" and "errors"
+            // in statsRes.requests and statsRes['500s']
+            Array.prototype.splice.apply(statsRes.requests,
```

Comment:
Wouldn't `statsRes.requests = requests` work just as well?

Reply:
Well that's what you do just below, but then why applying this to `statsRes.requests` beforehand?